internal/plugins/plugins.go

```diff
@@ -166,75 +180,55 @@ func createCmdRunE(pluginExecutable string) func(cmd *cobra.Command, args []stri
 
 // filterRootFlags filters out any flags that were parsed by the root command's persistent flags
 // and processes them for the host application, returning only plugin-specific args.
-func filterRootFlags(args []string, persistentFlags *pflag.FlagSet) ([]string, error) {
-	pluginArgs := []string{}
-
-	var (
-		previousArgFlagName     = ""
-		handlePreviousArgAsFlag = false
-	)
-
-	for _, arg := range args {
-		switch {
-		case handlePreviousArgAsFlag && previousArgFlagName != "":
-			err := persistentFlags.Set(previousArgFlagName, arg)
-			if err != nil {
-				return nil, fmt.Errorf("failed to set persistent flag '%s' with value '%s': %w", previousArgFlagName, arg, err)
-			}
-			handlePreviousArgAsFlag = false
-		case len(arg) > 0 && arg[0] == '-':
-			// The argument is a flag, remove leading dashes
-			flagArg := strings.TrimLeft(arg, "-")
-
-			// Handle flags in the format --flag=value
-			if strings.Contains(flagArg, "=") {
-				parts := strings.SplitN(flagArg, "=", 2)
-				flagName := parts[0]
-
-				if flag := persistentFlags.Lookup(flagName); flag != nil {
-					err := persistentFlags.Set(flagName, parts[1])
-					if err != nil {
-						return nil, fmt.Errorf("failed to set persistent flag '%s' with value '%s': %w", flagName, parts[1], err)
-					}
-				} else if len(flagName) == 1 && persistentFlags.ShorthandLookup(flagName) != nil {
-					flag := persistentFlags.ShorthandLookup(flagName)
-					err := persistentFlags.Set(flag.Name, parts[1])
-					if err != nil {
-						return nil, fmt.Errorf("failed to set persistent flag '%s' with value '%s': %w", flag.Name, parts[1], err)
-					}
-				} else {
-					pluginArgs = append(pluginArgs, arg)
-				}
-			} else {
-				if flag := persistentFlags.Lookup(flagArg); flag != nil {
-					if flag.Value.Type() == "bool" {
-						err := persistentFlags.Set(flagArg, "true")
-						if err != nil {
-							return nil, fmt.Errorf("failed to set persistent flag '%s' with value 'true': %w", flagArg, err)
-						}
-					} else {
-						previousArgFlagName = flagArg
-						handlePreviousArgAsFlag = true
-					}
-				} else if len(flagArg) == 1 && persistentFlags.ShorthandLookup(flagArg) != nil {
-					flag := persistentFlags.ShorthandLookup(flagArg)
-					if flag.Value.Type() == "bool" {
-						err := persistentFlags.Set(flag.Name, "true")
-						if err != nil {
-							return nil, fmt.Errorf("failed to set persistent flag '%s' with value 'true': %w", flag.Name, err)
-						}
-					} else {
-						previousArgFlagName = flag.Name
-						handlePreviousArgAsFlag = true
-					}
-				} else {
-					pluginArgs = append(pluginArgs, arg)
-				}
-			}
-		default:
+func filterRootFlags(cmd *cobra.Command, args []string) []string {
+	pluginArgs := make([]string, 0) // Initialize as an empty slice
+	rootFlags := cmd.Root().PersistentFlags()
+
+	// isRootFlag checks if a given argument (like "--profile") is a known persistent flag on the root command.
+	isRootFlag := func(arg string) *pflag.Flag {
```

Comment:
I might rename this, since it's more retrieving the root flag than running a boolean check

Reply:
changed to 'lookupRootFlag'